psydac/core/bsplines.py

```diff
@@ -225,6 +228,11 @@ def basis_funs_all_ders( knots, degree, x, span, n ):
     # Derivatives higher than degree are = 0.
     ne = min( n, degree )
 
+    # Normalization to get M-Splines
+    if normalization == 'M':
+        scaling = 1 / np.array([(knots[i + degree + 1] - knots[i]) / (degree + 1)
+                                for i in range(span - degree, span + 1)])
+
```

Comment:
It does not seem necessary to store this temporary array. See below...

Reply:
Fixed in [3d32ea6](https://github.com/pyccel/psydac/pull/193/commits/3d32ea6ea706b76e80e6d0f74ca8acf36c47be3b)